app/src/main/java/com/expidev/gcmapp/json/TrainingJsonParser.java

```diff
@@ -32,15 +33,23 @@
     @NonNull
     public static Training parseTraining(@NonNull final JSONObject json) throws JSONException {
         final Training training = new Training();
-        training.setId(json.getInt("id"));
+        if (json.has("id")) training.setId(json.getInt("id"));
```

Comment:
you may want to use `json.optInt("Id")` instead.

Something else I just noticed is that the id is currently an int, in order for it to be used by internal android functions it should be a long.